packages-native/actions/src/camera/TakePictureAdvanced.ts

```diff
@@ -0,0 +1,225 @@
+                        resultObject.set("Width", response.width);
+                        resultObject.set("Height", response.height);
+                        resultObject.set("FileName", response.fileName);
+                        resultObject.set("FileSize", response.fileSize);
+                        resultObject.set("FileType", response.type);
+                        resolve(resultObject);
+                    });
+                })
+                .catch(error => {
+                    if (error === "canceled") {
+                        resolve(resultObject);
+                    } else {
+                        throw new Error(error);
+                    }
+                });
+        });
+    });
+
+    function takePicture(): Promise<string | undefined> {
+        return new Promise((resolve, reject) => {
+            const options = getOptions();
+            const method = getPictureMethod();
+
+            method(options, response => {
+                if (response.didCancel) {
+                    return resolve();
+                }
+
+                if (response.error) {
+                    const unhandledError = handleImagePickerError(response.error);
+                    if (!unhandledError) {
+                        return resolve();
+                    }
+                    return reject(new Error(response.error));
+                }
+
+                return resolve(response.uri);
+            });
+        });
+    }
+
+    function storeFile(imageObject: mendix.lib.MxObject, uri: string): Promise<boolean> {
+        return new Promise((resolve, reject) => {
+            fetch(uri)
+                .then(res => res.blob())
+                .then(blob => {
+                    const guid = imageObject.getGuid();
+                    // eslint-disable-next-line no-useless-escape
+                    const filename = /[^\/]*$/.exec(uri)![0];
+                    const onSuccess = (): void => resolve(true);
+                    const onError = (error: Error): void => reject(error);
+
+                    mx.data.saveDocument(guid, filename, {}, blob, onSuccess, onError);
+                });
+        });
+    }
+
+    // eslint-disable-next-line @typescript-eslint/explicit-function-return-type
+    function getPictureMethod() {
+        const source = pictureSource ? pictureSource : "either";
+
+        switch (source) {
+            case "imageLibrary":
+                return ImagePicker.launchImageLibrary;
+            case "camera":
+                return ImagePicker.launchCamera;
+            case "either":
+            default:
+                return ImagePicker.showImagePicker;
+        }
+    }
+
+    // eslint-disable-next-line @typescript-eslint/explicit-function-return-type
+    function getOptions() {
+        const { maxWidth, maxHeight } = getPictureQuality();
+
+        return {
+            mediaType: "photo" as "photo",
```

Comment:
??

Reply:
This bit I took from the original TakePicture. I want to keep the two the same where possible.